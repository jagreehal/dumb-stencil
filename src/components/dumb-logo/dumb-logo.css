:host {
  display: inline;
}

.c-logo__img {
  display: block;
}
