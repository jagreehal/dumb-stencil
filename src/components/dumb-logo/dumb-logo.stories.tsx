import '@stencil/core';

import { storiesOf } from '@storybook/html';

const stories = storiesOf('Global/Logo', module);

stories.add('Default', () => {
  return `<dumb-logo
      href="http://bradfrost.com"
      src="http://via.placeholder.com/120x60"
      alt="Company Name" />`;
});
