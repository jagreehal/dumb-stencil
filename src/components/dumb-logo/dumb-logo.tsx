import { Component, Prop } from '@stencil/core';

@Component({
  tag: 'dumb-logo',
  styleUrl: 'dumb-logo.css',
  shadow: true
})
export class DumbLogo {
  @Prop() alt: string;
  @Prop() href: string;
  @Prop() src: string;

  render() {
    return (
      <a href={this.href} class="c-logo" rel="home">
        <img class="c-logo__img" src={this.src} alt={this.alt} />
      </a>
    );
  }
}
