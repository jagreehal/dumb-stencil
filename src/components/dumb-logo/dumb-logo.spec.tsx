import { TestWindow } from '@stencil/core/testing';
import { DumbLogo } from './dumb-logo';

describe('dumb-logo', () => {
  let element;
  let window;

  beforeEach(async () => {
    window = new TestWindow();
    element = await window.load({
      components: [DumbLogo],
      html: '<dumb-logo></dumb-logo>'
    });
  });

  it('should match snapshot', async () => {
    expect(element).toMatchSnapshot();
  });
});
