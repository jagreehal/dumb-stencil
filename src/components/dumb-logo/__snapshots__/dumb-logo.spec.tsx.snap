// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dumb-logo should match snapshot 1`] = `
<dumb-logo
  class="hydrated"
>
  
  <a
    class="c-logo"
    rel="home"
  >
    <img
      class="c-logo__img"
    />
  </a>
</dumb-logo>
`;
